### PR TITLE
Check go version

### PR DESCRIPTION
**Pull request by fraenkel**
This does not allow one to use go from tip which has a version of "devel". Wasn't sure if we want to support this.

When I do a dep ensure, a bunch of changes occur in the root Gopkg.toml and vendor directory. Wasn't sure how it was being managed, but I tried to keep it to a minimal change.

### CODE REVIEW

**Comment by droot** [comment]
@fraenkel Thanks for the PR.

The repo has two vendors:

- One at the root of the repo which is for project which vendors `kubebuilder` repo as lib
- One under `cmd` which is for all the `kubebuilder` cmd binary.

For this PR, since the change is under `cmd`, we want to update the vendor under `cmd`. So you will have to change to `cmd` dir and `dep ensure -add <semvar-repo-path>`.

File: cmd/kubebuilder/initproject/init.go [hunk 1]
```diff
@@ -78,9 +77,24 @@ kubebuilder init repo --domain mydomain
...
+		if err != nil {
+			log.Fatalf("Invalid go version %q: %v", goVersion, err)
+		}
+		c, err := semver.NewConstraint(">= 1.10")
+		if err != nil {
+			log.Fatal("Invalid constraint: %v", err)
+		}
+		if !c.Check(ver) {
+			log.Fatalf("The go version is %v, must be 1.10+", goVersion)
+		}
```
Comment (droot) [inline cmd/kubebuilder/initproject/init.go]: Can we abstract the above code into a separate function ?

File: cmd/kubebuilder/initproject/init.go [hunk 2]
```diff
@@ -78,9 +77,24 @@ kubebuilder init repo --domain mydomain
...
-	version := runtime.Version()
-	if versionCmp(version, "go1.10") < 0 {
-		log.Fatalf("The go version is %v, must be 1.10+", version)
+	cmd := exec.Command("go", "version")
+	out, err := cmd.Output()
+	if err != nil {
+		log.Fatalf("Could not execute 'go version': %v", err)
+	}
+
+	goVersion := strings.TrimPrefix(strings.Split(string(out), " ")[2], "go")
```
Comment (droot) [inline cmd/kubebuilder/initproject/init.go]: We can get invalid index error here.

File: cmd/kubebuilder/initproject/init.go (hunk 1 above)
Comment (fraenkel) [reply cmd/kubebuilder/initproject/init.go]: ack

File: cmd/kubebuilder/initproject/init.go (hunk 2 above)
Comment (fraenkel) [reply cmd/kubebuilder/initproject/init.go]: ack

**Comment by fraenkel** [comment]
Unrelated TestStorageBee failure

**Pull request merged**